src/common/store/reblogs/index.ts

```diff
@@ -1,35 +1,41 @@
-import { Dispatch } from "redux";
+import {Dispatch} from "redux";
 
-import { Reblog, Actions, ActionTypes, ReloadAction } from "./types";
+import {Reblog, Actions, ActionTypes, ReloadAction} from "./types";
 
 import * as ls from "../../util/local-storage";
 
 const load = (): Reblog[] => {
-  return ls.get("reblogs") || [];
+    return ls.get("reblogs") || [];
 };
 
 export const initialState: Reblog[] = [];
 
 export default (state: Reblog[] = initialState, action: Actions): Reblog[] => {
-  switch (action.type) {
-    case ActionTypes.RELOAD: {
-      return load();
+    switch (action.type) {
+        case ActionTypes.RELOAD: {
+            return load();
+        }
+        default:
+            return state;
     }
-    default:
-      return state;
-  }
 };
 
 /* Actions */
 export const addReblog = (account: string, author: string, permlink: string) => (dispatch: Dispatch) => {
-  const data: Reblog[] = [...(ls.get("reblogs") || []), { account, author, permlink }];
-  ls.set("reblogs", data);
-  dispatch(reloadAct());
+    const data: Reblog[] = [...(ls.get("reblogs") || []), {account, author, permlink}];
```

Comment:
here cache we get from Feed page right? If post is opened directly with URL, this doesn't show user about reblog state, let's change that, there is already special call to get list of reblogs from api. 

`{"jsonrpc":"2.0", "method":"condenser_api.get_reblogged_by", "params":["good-karma","hivesigner-ui-simplified-import-and-login"], "id":1}`
This we can use when post is opened and not matter user state.

Reply:
we can save one request, if user coming from feeds by using this cache method still, but if coming from direct link, then we will have to do 1 more request to get reblogs list.

Reply:
> here cache we get from Feed page right? If post is opened directly with URL, this doesn't show user about reblog state, let's change that, there is already special call to get list of reblogs from api.
> 
> `{"jsonrpc":"2.0", "method":"condenser_api.get_reblogged_by", "params":["good-karma","hivesigner-ui-simplified-import-and-login"], "id":1}`
> This we can use when post is opened and not matter user state.

We store that cache in local storage. No matter page, that cache loaded from local storage to store on page load and updated once the user reblogs or deletes reblog.

Reply:
I tested above request, it seem to work quite fast so we might as well save user some storage if it works good. On feeds page we already have reblogged_by field and in post page, we can make 1 more request to fetch list. On mobile app, we will use this as well, instead of internal API we created for reblogs.

Reply:
I think you are referring to getting reblog count of a specific post. But this file (store/reblogs/index.ts) is responsible to manage reblogs that the user (or users) make. So we show the user if they reblogged a post (on post lists and post detail) or not.

Reply:
Right, with this call we can get count as well, as length of array. And know if user reblogged without caching. So I am thinking combining both solution seem good option. Or not cache reblogs at all. Which one you think is good?

Reply:
Is there an endpoint to get posts that the user reblogged?

Reply:
@talhasch I don't think it exist. Mostly post specific reblog info with above api call I shared `condenser_api.get_reblogged_by`

Reply:
@feruzm changed local storage caching to `condenser_api.get_blog_entries`

Reply:
Integrating `condenser_api.get_reblogged_by` to show rebloggers....